src/physics/lsm_driver.f90

```diff
@@ -797,9 +824,138 @@ subroutine lsm_init(domain,options)
+            ! MMINLU  = options%lsm_options%LU_Categories !"MODIFIED_IGBP_MODIS_NOAH"
+            ISLAKE  = options%lsm_options%lake_category
+
+            ! allocate_noah_data already sets xice_threshold, so if we are using noah (mp/lsm) leave as is.
+            if(.not.(options%physics%landsurface==kLSM_NOAHMP .OR. options%physics%landsurface==kLSM_NOAH)) then
+                xice_threshold = 1.0  ! allocate_noah_data sets it to 1., BUT WRF's module_physics_init.F sets xice_threshold to 0.5 .... so?
+            endif
+
+            lake_count=0
+            if(ISLAKE==-1) then
+                if(this_image()==1) write(*,*)  "   WARNING: no lake category in LU data: The model will try to guess lake-gridpoints. This option has not been properly tested!"
+                lakeflag=0  ! If no lake cat is provided, the lake model will determine lakes based
+                            ! on the criterion (ivgtyp(i,j)==iswater .and. ht(i,j)>=lake_min_elev))
+            else
+                lakeflag=1
+                ! from WRF's module_initialize_real.F:
+                DO j = jts, MIN(jde-1,jte)
+                    DO i = its, MIN(ide-1,ite)
+                    !    IF ( grid%lu_index(i,j) .NE. grid%islake ) THEN
+                        if(domain%veg_type(i,j) .NE. ISLAKE ) then
+                            lakemask(i,j) = 0       ! grid%lakemask(i,j) = 0
+                        ELSE
+                            lakemask(i,j) = 1       ! grid%lakemask(i,j) = 1
+                            lake_count= lake_count + 1
+                        end if
+                    END DO
+                END DO
+            endif
+            ! if(options%parameters%debug) write(*,*)"   ",lake_count, " lake cells in image ", this_image()
+
+            ! setlake_depth_flag and use_lakedepth flag. (They seem to be redundant, but whatever):
+            if( associated(domain%lake_depth%data_2d) ) then
+                if(this_image()==1) write(*,*) "   Using Lake depth data "
+                use_lakedepth = 1
+                lake_depth_flag = 1
+            else
+                use_lakedepth = 0
+                lake_depth_flag = 0
+            endif
+
+            call lakeini( &
+                IVGTYP = domain%veg_type                        &
+                ,ISLTYP = domain%soil_type                      &
+                ,HT=domain%terrain%data_2d                      & ! terrain height [m] if ht(i,j)>=lake_min_elev -> lake  (grid%ht in WRF)
+                ,SNOW=domain%snow_water_equivalent%data_2d      & !i  ! SNOW in kg/m^2  (NoahLSM: SNOW liquid water-equivalent snow depth (m)
+                ,lake_min_elev=5.                               & ! minimum elevation of lakes. May be used to determine whether a water point is a lake in the absence of lake category. If the landuse type includes 'lake' (i.e. Modis_lake and USGS_LAKE), this variable is of no effects.
+                ,restart=.false.                                &
```

Comment:
What will this do for ICAR restarts? do we need to read in some of the above allocated variables (and write them to restart files in the first place?)

Reply:
Yes that would be nice. I just figured since Noah and Noah MP arent set up for restarts, so shouldnt this. But I will look into it.

Reply:
Thanks, the noah's both save SWE, soil moisture, soil temperature (and canopy water/ice?), for restarts, those are the most critical variables.  Same here, you don't necessarily need every single variable, if there are a few that change from second to second, but anything that takes more than a few time steps to spin up should be saved... and it would be better to just save everything that is a state variable (I think NoahMP saves pretty much all states, I might need to look through NoahLSM again.)